replica/database.hh

```diff
@@ -445,6 +444,7 @@ private:
 
     // Provided by the database that owns this commitlog
     db::commitlog* _commitlog;
+    bool _readonly;
```

Comment:
Please comment on this flag and its relationship with commitlog

Reply:
Can we actually use `_commitlog == nullptr" as readonly indication?